staging/src/k8s.io/client-go/applyconfigurations/core/v1/pod.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by applyconfiguration-gen. DO NOT EDIT.
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	v1 "k8s.io/client-go/applyconfigurations/meta/v1"
+)
+
+// PodApplyConfiguration represents an declarative configuration of the Pod type for use
+// with apply.
+type PodApplyConfiguration struct {
+	v1.TypeMetaApplyConfiguration   `json:",inline"`
+	v1.ObjectMetaApplyConfiguration `json:"metadata,omitempty"`
+	Spec                            *PodSpecApplyConfiguration   `json:"spec,omitempty"`
+	Status                          *PodStatusApplyConfiguration `json:"status,omitempty"`
+}
+
+// PodApplyConfiguration constructs an declarative configuration of the Pod type for use with
+// apply.
+func Pod(name, namespace string) *PodApplyConfiguration {
+	b := &PodApplyConfiguration{}
+	b.SetName(name)
+	b.SetNamespace(namespace)
+	b.SetKind("Pod")
+	b.SetAPIVersion("v1")
+	return b
+}
+
+// SetKind sets the Kind field in the declarative configuration to the given value.
+func (b *PodApplyConfiguration) SetKind(value string) *PodApplyConfiguration {
+	b.Kind = &value
+	return b
+}
+
+// SetAPIVersion sets the APIVersion field in the declarative configuration to the given value.
+func (b *PodApplyConfiguration) SetAPIVersion(value string) *PodApplyConfiguration {
+	b.APIVersion = &value
+	return b
+}
+
+// SetName sets the Name field in the declarative configuration to the given value.
+func (b *PodApplyConfiguration) SetName(value string) *PodApplyConfiguration {
```

Comment:
Shouldn't this be part of `metadata`?

Reply:
It is. I'm following the convention that the go structs take where `ObjectMeta`is embedded but not json inlined:

```
v1.ObjectMeta `json:"metadata,omitempty"`
```

This results in the Name field being on the top-level type but being serialized under "metadata".

I didn't see any reason to break from the go struct convention.